unit_tests/src/catch2_compat_include/catch2/catch_all.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef IRODS_CATCH2_V2_SHIM_H
+#define IRODS_CATCH2_V2_SHIM_H
+
+#include <catch2/catch.hpp>
+
+// NOLINTBEGIN(misc-unused-alias-decls)
+
+namespace Catch
+{
+    namespace Matchers
+    {
+
+        template <typename... Args>
+        constexpr auto ContainsSubstring(Args&&... args) -> decltype(Contains(std::forward<Args>(args)...))
+        {
+            return Contains(std::forward<Args>(args)...);
+        }
+
+        using Impl::MatcherBase;
+    } //namespace Matchers
+
+    namespace clara { }
```

Comment:
Could remove space between braces and wrap in `// clang-format off/on`.

Up to you.